### DISCUSSION TITLE
Use Case: Seed A Full Sync to Remote Host

### DISCUSSION THREAD

**Discussion by rotor-head** [zrepl/zrepl#569] [Q&A] [answered]
Use Case Scenario:

> 1. Remote Server(s) with over 2TB of data each.
> 2. Central Server for backups.
> 3. Standard Public Internet (read slow links) between Remote and Central backup.

Looking over the examples, it appears the use case of a laptop to external drive could serve as a template to provide the initial data seed of a full pool backup. Then, transfer the data off the external drive (directly connected) to the Central backup to server as the initial full dataset for the Remote host to begin creating and sending snapshots against.

Are there any limitations that would prevent this scenario from working? If no, are there any unique configurations needed to ensure success?

Forgive me if this is such a simple scenario - I'm new to ZFS replication/zrepl and learning on the fly here.

**Comment by problame** [comment, marked as answer]
Sorry for the late reply.
So, if I understand correctly, you want to do the initial transfer to an external HDD, then move that HDD to the central backup server, and continue (?) using it there for incremental replication?

That would be possible.
Assuming you want to use `push` style setup, do it as follows:

1. On the remote site, setup a pair of `push` and `sink` job, with transport type `local`. Use that to do the initial replication. Keep in mind that zrepl [currently always uses (only) the latest snapshot for initial replication](https://github.com/zrepl/zrepl/pull/470).
2. After initial replication is done, stop zrepl, comment out the local `sink`, and export+disconnect the pool on the external HDD.
3. Move the HDD to the central backup server and import the zpool on it.
4. On the central backup server, define a `sink` job, and switch over the remote site's `push` job to connect to that `sink` instead of the local one.
  1. Make sure that the client identity stays the same ([read up here](https://zrepl.github.io/configuration/overview.html?highlight=client%20identity#how-the-passive-side-works)) 
5. Start the zrepl jobs again, it should pick up incremental replication.

Please report whether this solved your problem. We might turn it into a runbook, similar to the one we have for [moving the sending side](https://zrepl.github.io/usage/runbooks/migrating_sending_side_to_new_zpool.html)

**Reply by rotor-head** [reply]
Yes, you deciphered my use case precisely. The HDD is the initial seed of data to avoid transmitting large quantities of data over the public wire - where bandwidth limits would take several days for the first job to complete.

It will be a few weeks before I'll be in a spot to work through the steps. I'm sure there are others who would benefit from a runbook and happy to help.
